XouDevSpec/Classes/Model/ApiCallerModel.swift

```diff
@@ -28,8 +29,12 @@ public struct ApiCallerModel {
     
     
     public  func getAnimeData(completetionHandler: @escaping(Result<[AnimeDetails], AnimeError>) -> Void) {
+        
+        guard let url = URL(string: "https://jikan1.p.rapidapi.com/search/anime?q=\(searchText)") else {
+            return
```

Comment:
You can also call the completion handler and pass in the AnimeError for ProblemWithUrl, instead of just the return.